File: C#/Double Circular/DoubleCircular.cs
using System;

public class Node
{
    public int Data;
    public Node Next;
    public Node Prev;

    public Node(int data)
    {
        Data = data;
        Next = null;
        Prev = null;
    }
}

public class DoublyCircularLinkedList
{
    public Node Head;

    public DoublyCircularLinkedList()
    {
        Head = null;
    }

    public void Insert(int data)
    {
        Node newNode = new Node(data);
        if (Head == null)
        {
            Head = newNode;
            newNode.Next = Head;
            newNode.Prev = Head;
        }
        else
        {
            Node last = Head.Prev;
            last.Next = newNode;
            newNode.Next = Head;
            newNode.Prev = last;
            Head.Prev = newNode;
        }
    }

    public void DeleteNode(int key)
    {
        if (Head == null) return;

        Node current = Head;

        // If the head node holds the key
        if (current.Data == key)
        {
            if (current.Next == Head)
            {
                Head = null;
                return;
            }

            Node last = Head.Prev;
            last.Next = current.Next;
            current.Next.Prev = last;
            Head = current.Next;
            return;
        }

        // Search for the node to be deleted
        while (current.Next != Head && current.Data != key)
        {
            current = current.Next;
        }

        if (current.Data == key)
        {
            current.Prev.Next = current.Next;
            current.Next.Prev = current.Prev;
        }
    }

    public void Display()
    {
        if (Head == null) return;
        Node current = Head;
        do
        {
            Console.Write(current.Data + " <-> ");
            current = current.Next;
        } while (current != Head);
        Console.WriteLine("(head)");
    }
}

class Program
{
    static void Main()
    {
        DoublyCircularLinkedList list = new DoublyCircularLinkedList();
        list.Insert(3);
        list.Insert(2);
        list.Insert(1);

        Console.WriteLine("Original List:");
        list.Display();

        list.DeleteNode(2);
        Console.WriteLine("After Deleting 2:");
        list.Display();
    }
}

File: JavaScript/Double Circular/DoubleCircular.js
class Node {
    constructor(data) {
        this.data = data;
        this.next = null;
        this.prev = null;
    }
}

class DoublyCircularLinkedList {
    constructor() {
        this.head = null;
    }

    insert(data) {
        const newNode = new Node(data);
        if (this.head === null) {
            this.head = newNode;
            newNode.next = newNode;
            newNode.prev = newNode;
        } else {
            const last = this.head.prev;
            last.next = newNode;
            newNode.next = this.head;
            newNode.prev = last;
            this.head.prev = newNode;
        }
    }

    deleteNode(key) {
        if (this.head === null) return;

        let current = this.head;

        // If the head node holds the key
        if (current.data === key) {
            if (current.next === this.head) {
                this.head = null;
                return;
            }

            const last = this.head.prev;
            last.next = current.next;
            current.next.prev = last;
            this.head = current.next;
            return;
        }

        // Search for the node to be deleted
        while (current.next !== this.head && current.data !== key) {
            current = current.next;
        }

        if (current.data === key) {
            current.prev.next = current.next;
            current.next.prev = current.prev;
        }
    }

    display() {
        if (this.head === null) return;
        let current = this.head;
        do {
            process.stdout.write(current.data + ' <-> ');
            current = current.next;
        } while (current !== this.head);
        console.log('(head)');
    }
}

// Example usage
const list = new DoublyCircularLinkedList();
list.insert(3);
list.insert(2);
list.insert(1);

console.log('Original List:');
list.display();

list.deleteNode(2);
console.log('After Deleting 2:');
list.display();
